fedimint-client/src/sm/executor.rs

```diff
@@ -45,6 +47,34 @@ pub(crate) enum ExecutorDbPrefixes {
     InactiveStates = 0xa2,
 }
 
+#[derive(Serialize, Deserialize)]
+pub struct StateMachineCreated {
+    operation_id: OperationId,
+    module_id: ModuleInstanceId,
+}
+
+impl Event for StateMachineCreated {
+    const MODULE: Option<fedimint_core::core::ModuleKind> = None;
+
+    const KIND: EventKind = EventKind::from_static("sm-created");
+    const TRIMABLE: bool = true;
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct StateMachineUpdated {
+    operation_id: OperationId,
+    terminal: bool,
+    outcome: serde_json::Value,
```

Comment:
This seems to be the outcome of the trigger future, but that's kinda pointless to know unless you already know the previous state. I'd rather include the state in both `StateMachineUpdated` and `StateMachineCreated` and would consider merging the two. That way the `subscribe_*_updates` functions could already switch to the event log in theory, getting rid of the broadcast channel currently used.

Reply:
@elsirion `DynState` seems encodable, but not serializable to json. Do you mean to serialize it to e.g. hex? Is this useful for anyone?

Reply:
@elsirion So no state, no outcome? It's OK, adding more fields is backward compat.

Reply:
Exactly, let's see how far that gets us and add more things as needed.